pkg/securitycontext/provider.go

```diff
@@ -95,3 +96,53 @@ func makeCapabilites(capAdd []api.Capability, capDrop []api.Capability) ([]strin
 	}
 	return addCaps, dropCaps
 }
+
+func determineEffectiveSecurityContext(pod *api.Pod, container *api.Container) *api.SecurityContext {
```

Comment:
That is correct.

On Tue, Sep 29, 2015 at 6:05 AM, Paul Morie <notifications@github.com>
wrote:

> In pkg/securitycontext/provider.go
> <https://github.com/kubernetes/kubernetes/pull/14705#discussion_r40632969>
> :
>
> > @@ -95,3 +96,53 @@ func makeCapabilites(capAdd []api.Capability, capDrop []api.Capability) ([]strin
> >  	}
> >  	return addCaps, dropCaps
> >  }
> > +
> > +func determineEffectiveSecurityContext(pod *api.Pod, container *api.Container) *api.SecurityContext {
>
> @pweil- <https://github.com/pweil-> @smarterclayton
> <https://github.com/smarterclayton> @liggitt <https://github.com/liggitt>
> I'm starting to think this is wrong. I think if you have
> container.SecurityContext specified at all, it should win for that
> container over pod.Spec.SecurityContext, period. Otherwise old clients
> cannot reason correctly about what a container's SecurityContext is.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14705/files#r40632969>.
>